apps/backend-functions/src/negotiation.ts

```diff
@@ -1,75 +1,121 @@
+      .where('type', '==', 'sale');
+    const offerContractsSnap = await tx.get(offerContractsQuery);
+    const negotiationSnaps = await Promise.all(offerContractsSnap.docs.map(doc => lastNegotiation(doc.id)));
+
+    const contractsStatus: ContractStatus[] = negotiationSnaps.map(nego => {
+      const _meta = formatDocumentMetaFromFirestore(nego._meta);
+      const initial = nego.initial.toDate();
+      const isPending = nego.status === 'pending';
+      if (isInitial({ _meta, initial }) && isPending) return 'pending';
+      if (isPending) return 'negotiating';
+      return nego.status;
+    });
+
+    let newOfferStatus = offer.status;
+    const negotiatingStatuses = ['negotiating', 'accepted', 'declined'];
+    const acceptedStatuses = ['accepted', 'declined'];
+    if (contractsStatus.some(status => negotiatingStatuses.includes(status))) newOfferStatus = 'negotiating';
+    if (contractsStatus.every(status => acceptedStatuses.includes(status))) newOfferStatus = 'accepted';
+    if (contractsStatus.every(status => status === 'declined')) newOfferStatus = 'declined';
+    if (contractsStatus.every(status => status === 'pending')) newOfferStatus = 'pending';
+
+    if (newOfferStatus === offer.status) return;
+    return tx.update(offerRef, { status: newOfferStatus });
+  })
+}
 
 export async function onNegotiationCreated(negotiationSnapshot: FirebaseFirestore.DocumentSnapshot<Negotiation<Timestamp>>) {
   const negotiation = negotiationSnapshot.data();
   const _meta = formatDocumentMetaFromFirestore(negotiation._meta);
   const initial = negotiation.initial.toDate();
 
-  if (isInitial({ _meta, initial })) return;
-
   const path = negotiationSnapshot.ref.path;
   // contracts/{contractId}/negotiations/{negotiationId}
   const contractId = path.split('/')[1]
-
-
-  const stakeholders = negotiation.stakeholders.filter(stakeholder => {
-    return ![negotiation.createdByOrg, centralOrgId.catalog].includes(stakeholder)
-  });
-
-  if (!stakeholders.length) return;
-
-  const getNotifications = (org: Organization) => org.userIds.map(userId => createNotification({
+  const contract = await getDocument<Sale>(`/contracts/${contractId}`);
+
+  //Send notification to seller about an offer made on his title.
+  if (isInitial({ _meta, initial })) {
+    const stakeholders = negotiation.stakeholders.filter(stakeholder => {
+      return ![contract.buyerId, centralOrgId.catalog].includes(stakeholder);
+    });
+
+    if (!stakeholders.length) return;
+
+    const getNotifications = (org: Organization) => org.userIds.map(userId => createNotification({
+      toUserId: userId,
+      type: 'contractCreated',
+      docId: contractId,
+      docPath: path,
+      _meta: createDocumentMeta({ createdFrom: 'catalog' })
+    }));
+    const promises = stakeholders.map(stakeholder =>
+      getDocument<Organization>(`orgs/${stakeholder}`)
+        .then(getNotifications)
+        .then(triggerNotifications)
+    );
+    return await Promise.all(promises);
```

Comment:
you only need to return
```suggestion
    return Promise.all(promises);
```